### ISSUE TITLE
log simple metrics

### ISSUE THREAD

**Issue by Fersoil** [swiss-ai/amr-fma#6]
## Metrics and observability during training

Currently `lora_sft.py` only reports whatever TRL/Hugging Face Trainer logs by default.
This issue tracks adding richer in-training observability before we start Phase 1 runs.

### What TRL already logs automatically

When `report_to="wandb"` is set, the following are sent to wandb out of the box:

- `train/loss` — cross-entropy loss per logging step
- `train/learning_rate` — the full LR schedule curve (useful to verify warmup)
- `train/grad_norm` — gradient norm; the first signal of instability
- `train/epoch`, `train/global_step`

### What we should add

**A `MetricsCallback`** in `lora_sft.py` (separate from `ManifestCallback`) that logs
on each `on_log` event:

| Metric | Why it matters |
|---|---|
| `train/perplexity` | More interpretable than raw loss; easy to compare across runs and model families |
| `system/gpu_memory_gb` | Track peak memory per step; critical for 32B runs in Phase 3 |
| `system/tokens_per_second` | Compute efficiency on GH200s; needed to validate GPU-hour estimates |

**An evaluation split** passed to `SFTTrainer` as `eval_dataset`. A small fixed held-out
slice (e.g. 512 examples) is enough to get an `eval/loss` curve alongside `train/loss`.
This is especially important for Phase 1 where we are deliberately inducing domain drift
and need to see the divergence clearly.

**Manifest upload as a wandb Artifact** at the end of `train()`:

```python
artifact = wandb.Artifact(
    name=f"manifest-{config.run.experiment_name}",
    type="manifest",
)
artifact.add_file(str(run_paths.manifest_path))
wandb.log_artifact(artifact)
```

This gives every run a versioned, downloadable record of its hyperparameters and
checkpoint registry, linked directly to the wandb run. Upload it once at the start
(planned config) and once at the end (with the checkpoint list filled in).

### Scope

All changes are confined to `amr_fma/fma/lora_sft.py`. No new dependencies — `wandb`
and `torch` are already in the environment.
